src/main/java/org/apache/beam/sdk/io/gcp/healthcare/WebPathParser.java

```diff
@@ -0,0 +1,46 @@
+package org.apache.beam.sdk.io.gcp.healthcare;
+
+import java.io.IOException;
+
+public class WebPathParser {
+
+    public static class DicomWebPath {
+        public String studyId;
+        public String seriesId;
+        public String instanceId;
+        public String dicomStorePath;
+        public String project;
+        public String location;
+        public String dataset;
+        public String storeId;
+    }
+
+    public DicomWebPath parseDicomWebpath(String unparsedWebpath) throws IOException {
+        String[] webPathSplit = unparsedWebpath.split("/dicomWeb/");
+
+        if (webPathSplit.length != 2) {
+            throw new IOException("Invalid DICOM web path");
+        }
+
+        DicomWebPath dicomWebPath = new DicomWebPath();
+
+        dicomWebPath.dicomStorePath = webPathSplit[0];
+        String[] storePathElements = dicomWebPath.dicomStorePath.split("/");
```

Comment:
You want to check the length of the Array here before accessing the indexes to be sure we never run into ArrayIndexOutOfBounds exception..